dev-support/ci/selective_ci_checks.sh

```diff
@@ -0,0 +1,470 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# shellcheck source=dev-support/ci/lib/_script_init.sh
+. dev-support/ci/lib/_script_init.sh
+
+# Parameter:
+#
+# $1 - COMMIT SHA of the incoming commit. If this parameter is missing, this script does not check anything,
+#      it simply sets all the version outputs that determine that all tests should be run.
+#      This happens in case the event triggering the workflow is 'schedule' or 'push'.
+#
+# The logic of retrieving changes works by comparing the incoming commit with the target branch
+# the commit addresses.
+#
+#
+declare -a pattern_array ignore_array
+
+ignore_array=(
+    "/README.md" # exclude root README
```

Comment:
This is confusing to me.  Is it trying to ignore all the readme's except the root readme?  so that means the tests would run if the root readme changes? why would we want that?

Reply:
README and other docs in root are already ignored  due to `SOURCES_TRIGGERING_TESTS`, which limits the list of "all changed files".  So we need to avoid subtracting it as "ignored", too.  I don't think currently any pattern array matches it, though.